skbio/tree/_tree.py

```diff
@@ -2430,10 +2432,14 @@ def update_result():
             starts, stops = [], []  # to calc ._start and ._stop for curr node
             for child in node.children:
                 if child.length is None:
-                    raise NoLengthError("Node with name '%s' doesn't have a "
-                                        "length." % child.name)
+                    warnings.warn("Node with name '%s' doesn't have a "
```

Comment:
Can you please update this warning message to explain that a length of zero will be used?